### PR TITLE
fix issue 15940 - ImplicitConversionTargets and class alias in struct

### PR DESCRIPTION
**Pull request by BBasile**

### CODE REVIEW

**Comment by JackStouffer** [comment]
LGTM

File: std/traits.d [hunk 1]
```diff
@@ -4125,6 +4125,13 @@ unittest
     static assert(is(ImplicitConversionTargets!(string)[0] == const(char)[]));
 }
 
+unittest // bugzilla 15940
+{
+    class Foo{}
+    struct Bar{Foo foo; alias foo this;}
+    static assert(ImplicitConversionTargets!(Bar).length == 0);
```
Comment (aG0aep6G) [inline std/traits.d]: But `Bar` does convert implicitly to `Foo`. So `ImplicitConversionTargets!Bar` should list `Foo`, shouldn't it? Although it looks like `ImplicitConversionTargets` doesn't recognize `alias this` at all.

Comment (BBasile) [inline std/traits.d]: > it looks like ImplicitConversionTargets doesn't recognize alias this at all.

Yes, it's was just working for a struct containing an alias this to a class (so the _bug_ fixed here), but all the other cases were not. for example:

```D
unittest // bugzilla 15940
{
    struct Bar{ubyte d; alias d this;}
    static assert(ImplicitConversionTargets!(Bar).length == 0);
}
```
passes. So the fix is conform will the global behavior of the template.

**Comment by berni44** [comment]
Are there any reasons, why this has never been merged?

**Pull request closed without merging**